STEP/TypeVisitor.cs

```diff
@@ -36,7 +36,7 @@ public class TypeVisitor : IVisitor {
         }
         else {
             n.Type.ActualType = TypeVal.Error;
-            throw new TypeException(TypeVal.Boolean, n.Left.Type.ActualType, n.Right.Type.ActualType);
+            throw new TypeException(n, TypeVal.Boolean, n.Left.Type.ActualType, n.Right.Type.ActualType);
```

Comment:
Ville det give mening at bruge `n.Left.Type` i stedet for `ActualType`? Hvis der f.eks. smides en exception fordi en type er constant, så er det jo en property på `Type`.

Reply:
ja, det er det - jeg lavede også en tostring på type, men det kræver bare lidt arbejde

Reply:
det hele bruger Type nu